### PR TITLE
[DYN-3782] Hide group style expander under debug mode

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Hides the Group Style expander if the debug mode Preferences Menu Debug Mode is enabled. Also sets this debug mode to off by default.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### CODE REVIEW

File: src/DynamoCore/Configuration/DebugModes.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ internal static void AddDebugMode(string name, string description, bool isEnable
         private static void RegisterDebugModes()
         {
             // Register app wide new debug modes here.
-            AddDebugMode("DynamoPreferencesMenuDebugMode", "Enable/Disable the Preferences MenuItem in the Dynamo menu.", true);
+            AddDebugMode("DynamoPreferencesMenuDebugMode", "Enable/Disable the Preferences MenuItem in the Dynamo menu.", false);
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/DebugModes.cs:50]: Can you update the description of the debug mode to `Enable/Disable the Preferences Panel new features in the Dynamo menu.`

**Review by QilongTang — CHANGES_REQUESTED** [review]
Once comment addressed, LGTM

File: src/DynamoCore/Configuration/DebugModes.cs (hunk 1 above)
Comment (Astul-Betizagasti) [reply src/DynamoCore/Configuration/DebugModes.cs]: Just pushed an update for this

**Review by QilongTang — APPROVED** [review]
LGTM

**Pull request merged**